WooCommerce/Classes/ViewRelated/Search/SearchViewController.swift

```diff
@@ -584,7 +603,7 @@ private extension SearchViewController {
     /// See `State` for the rules.
     ///
     func transitionToSyncingState() {
-        state = searchQuery.isEmpty ? stateIfSearchKeywordIsEmpty : .syncing
+        state = searchQuery.isEmpty && !searchUICommand.syncResultsWhenSearchQueryTurnsEmpty ? stateIfSearchKeywordIsEmpty : .syncing
```

Comment:
What do you think if we split this operator into declaring a variable first, just so is easier to follow. Something like:
```suggestion
let isEmptyOrShowingLoadStateWhenEmpty = 
	searchQuery.isEmpty &&
	!searchUICommand.syncResultsWhenSearchQueryTurnsEmpty

state = isEmptyOrShowingLoadStateWhenEmpty ? stateIfSearchKeywordIsEmpty : .syncing
```

Reply:
Good idea, done in 682f478cfcd7011ccc212897d9dcb09a77d972c0